core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsEndpointGroup.java

```diff
@@ -223,4 +230,18 @@ public String toString() {
                 .add("attemptsSoFar", attemptsSoFar)
                 .toString();
     }
+
+    class DnsQuestionListenerImpl implements DnsQuestionListener {
```

Comment:
Should we make this singleton? We can add a factory method for the default instance.
```java
// Returns the default instance.
DnsQuestionListener.of();
```
This might be useful to explain the default behavior in Javadoc when no custom `DnsQuestionListener` is configured.